### DISCUSSION TITLE
How to customize `outline`

### DISCUSSION THREAD

**Discussion by cafeclimber** [typst/typst#907] [Q&A] [answered]
Hi, just started attempting to write a template based on my university's dissertation requirements.

Looking at outline, I'm struggling to understand how to customize it really in any way. My first thought of a workaround was to simply only call the outline on specific portions I want separated by vertical space, however I then struggled to understand using a function in the target argument. Any thoughts?

As an example of the workaround, something like 

```
heading(outlined: false)[Table of Contents]
outline(title: none, indent: true, target: /* Front Matter */)
v(12pt)
outline(title: none, indent: true, target: /* Chapters */)
v(12pt)
outline(title: none, indent: true, target: /* Back Matter */)
pagebreak()
```

Ideally we could have more granular control in a show? Or if that's already possible, perhaps better documentation on customizing in this way.

(On another note, it might be good to state in the `outline` documentation that things can be hidden from the outline in the heading function. Took me a bit to find)

**Comment by LaurenzV** [comment]
Do you maybe have a picture of what you want the outline to look like in the end?

**Reply by cafeclimber** [reply]
Yeah so something like

```
List of Figures......................................ii
List of Tables.......................................iii
Acknowledgements ....................................iv

Chapter 1. Introduction .............................1
    Section title....................................1
    Other section....................................4
        Subsection...................................5

Chapter 2. Other stuff ..............................8
    Section title....................................9

Appendix A: Bits.....................................15

Appendix B: Bobs.....................................20
```

i.e. group front matter, then chapters and appendices etc single space in each individual context, but double space between them

**Comment by laurmaedje** [comment, marked as answer]
The `outline` is indeed not really customizable right now. You can write your own outline with `query` and do whatever there, but making the built-in outline easier to style is certainly something we should and will do.
